run_flax_speech_recognition_seq2seq_streaming_review.py

```diff
@@ -0,0 +1,1391 @@
+        batch[model_input_name] = inputs.get(model_input_name)[0]
+        batch["input_length"] = len(sample["array"])
+
+        # Process targets
+        input_str = batch[text_column_name].lower(
+        ) if do_lower_case else batch[text_column_name]
+        if do_remove_punctuation:
+            input_str = normalizer(input_str).strip()
+        batch["labels"] = tokenizer(input_str, truncation=True, max_length=max_label_length).input_ids
+        return batch
+
+    with training_args.main_process_first(desc="dataset map pre-processing"):
+        vectorized_datasets = raw_datasets.map(
+            prepare_dataset,
+            remove_columns=raw_datasets_features,
+        )
+
+    # Filter training data with inputs longer than max_input_length
+    def is_audio_in_length_range(length):
+        return min_input_length < length < max_input_length
+
+    if training_args.do_train:
+        vectorized_datasets["train"] = vectorized_datasets["train"].filter(
+            is_audio_in_length_range,
+            input_columns=["input_length"],
+        )
+
+    if training_args.do_eval:
+        vectorized_datasets["eval"] = vectorized_datasets["eval"].filter(
+            is_audio_in_length_range,
+            input_columns=["input_length"],
+        )
+
+    if training_args.do_predict:
+        vectorized_datasets["test"] = vectorized_datasets["test"].filter(
+            is_audio_in_length_range,
+            input_columns=["input_length"],
+        )
+
+    # Load metrics and write stats
+    metric_wer = evaluate.load("wer")
+    metric_cer = evaluate.load("cer")
+    do_normalize_eval = data_args.do_normalize_eval
+
+    def compute_metrics(pred_ids, label_ids):
+        # Replace padded labels by the padding token
+        for idx in range(len(label_ids)):
+            label_ids[idx][label_ids[idx] == -100] = tokenizer.pad_token_id
+
+        predictions = tokenizer.batch_decode(pred_ids, skip_special_tokens=True)
+        # We do not want to group tokens when computing the metrics
+        labels = tokenizer.batch_decode(label_ids, skip_special_tokens=True)
+
+        if do_normalize_eval:
+            pred_str = [normalizer(pred) for pred in predictions]
+            label_str = [normalizer(label) for label in labels]
+            # Filtering step to only evaluate the samples that correspond to non-zero references:
+            pred_str = [pred_str[i]
+                        for i in range(len(pred_str)) if len(label_str[i]) > 0]
+            label_str = [label_str[i]
+                         for i in range(len(label_str)) if len(label_str[i]) > 0]
+        else:
+            pred_str = predictions
+            label_str = labels
+
+        wer = 100 * metric_wer.compute(predictions=pred_str, references=label_str)
+        cer = 100 * metric_cer.compute(predictions=pred_str, references=label_str)
+        return {"wer": wer, "cer": cer}
+
+    def update_training_state(training_state, train_metrics, eval_metrics, step):
```

Comment:
(nit) This function and the proceeding one (`write_metric`) can probably go outside the `__main__` body

Reply:
If we move it outside, the function needs to somehow get access to the tokenizer, which is now in the scope.

Reply:
If we move it outside, the function needs to somehow get access to the tokenizer, which is now in the scope.

Reply:
Ah ok I see, would it work adding `tokenizer` as an argument to this function? If it requires the tokenizer happy to do it either way.